### PR TITLE
[llvm] use new noncopyable infrastructure

### PR DESCRIPTION
**Pull request by kavon**
companion to https://github.com/apple/swift/pull/72276

### CODE REVIEW

**Comment by kavon** [comment]
@swift-ci test

**Comment by kavon** [comment]
https://github.com/apple/swift/pull/72276
@swift-ci test

**Comment by kavon** [comment]
https://github.com/apple/swift/pull/72276
@swift-ci test

**Comment by kavon** [comment]
`urllib.error.HTTPError: HTTP Error 403: rate limit exceeded`

**Comment by kavon** [comment]
https://github.com/apple/swift/pull/72276
@swift-ci test

**Comment by kavon** [comment]
https://github.com/apple/swift/pull/72276
@swift-ci test

**Comment by kavon** [comment]
https://github.com/apple/swift/pull/72276
@swift-ci test

**Comment by kavon** [comment]
https://github.com/apple/swift/pull/72276
@swift-ci test

File: lldb/source/Plugins/TypeSystem/Swift/SwiftASTContext.cpp [hunk 1]
```diff
@@ -5616,11 +5615,20 @@ bool SwiftASTContext::GetProtocolTypeInfo(const CompilerType &type,
...
 
     unsigned num_witness_tables = 0;
+    unsigned num_protocols = 0;
     for (auto protoDecl : layout.getProtocols()) {
+      // Ignore invertible protocols like Copyable entirely.
+      if (protoDecl->getInvertibleProtocolKind())
+        continue;
+
+      num_protocols++;
+
```
Comment (adrian-prantl) [inline lldb/source/Plugins/TypeSystem/Swift/SwiftASTContext.cpp:5629]: This change LGTM!

**Comment by kavon** [comment]
https://github.com/apple/swift/pull/72276
@swift-ci test

**Comment by kavon** [comment]
https://github.com/apple/swift/pull/72276
@swift-ci test windows

**Comment by kavon** [comment]
https://github.com/apple/swift/pull/72276
@swift-ci test macOS

**Pull request merged**